src/main/java/org/example/service/jdbc/BuildingService.java

```diff
@@ -0,0 +1,238 @@
+package org.example.service.jdbc;
+
+import lombok.NoArgsConstructor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.example.dao.interfaces.IBuildingDAO;
+import org.example.dao.interfaces.IDeptDAO;
+import org.example.dao.interfaces.IRoomDAO;
+import org.example.dao.jdbc.BuildingDAO;
+import org.example.dao.jdbc.DeptDAO;
+import org.example.dao.jdbc.RoomDAO;
+import org.example.enums.EntityType;
+import org.example.model.Building;
+import org.example.model.Department;
+import org.example.model.Room;
+import org.example.service.exception.EntityNotFoundException;
+import org.example.service.exception.NoEntityCreatedException;
+import org.example.service.interfaces.IBuildingService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@NoArgsConstructor
+public class BuildingService implements IBuildingService {
+
+    private final IBuildingDAO buildingDAO = new BuildingDAO();
+    private final IDeptDAO deptDAO = new DeptDAO();
+    private final IRoomDAO roomDAO = new RoomDAO();
+    private final DepartmentService departmentService = new DepartmentService();
+    private static final Logger logger = LogManager.getLogger(BuildingService.class);
+
+    @Override
+    public Building getBuildingById(long id) throws EntityNotFoundException {
+        Building tempBuilding = getBasicBuildingById(id);
+        tempBuilding.setDepartments(getDeptsInBuilding(tempBuilding.getId()));
```

Comment:
This hould work i believe
```suggestion
        tempBuilding.setDepartments(getDeptsInBuilding(id));
```

Reply:
indeed)